keyboards/kyria/keymaps/winterNebs/keymap.c

```diff
@@ -0,0 +1,741 @@
+//       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+//                                  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+//     ),
+};
+
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+	return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+#ifdef ENCODER_ENABLE
+void encoder_update_user(uint8_t index, bool clockwise) {
+	if (index == 0) {
+		// Volume control
+		if (clockwise) {
+			tap_code(KC_RGHT);
+		} else {
+			tap_code(KC_LEFT);
+		}
+	}
+	else if (index == 1) {
+		// Page up/Page down
+		if (clockwise) {
+			tap_code(KC_DOWN);
+		} else {
+			tap_code(KC_UP);
+		}
+	}
+}
+#endif
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+#ifdef OLED_DRIVER_ENABLE
+bool left = false;
+bool right = false;
+bool lastl = false;
+bool lastr = false;
+#endif
+bool bksp = false;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	#ifdef CONSOLE_ENABLE
+	uprintf("KL: kc: %u, col: %u, row: %u, pressed: %u, total: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed, record->event.key.col + 10 * record->event.key.row);
+	#endif 
+	#ifdef OLED_DRIVER_ENABLE
+	if(record->event.pressed){
+		uint8_t n = record->event.key.col + 10 * record->event.key.row;
+		if (n<40) {
+			left = true;
+		} 
+		else {
+			right = true;
+		}
+	}
+	#endif
+	switch (keycode) {
+		case QWERTY:
+			if (record->event.pressed) {
+				set_single_persistent_default_layer(_QWERTY);
+			}
+			return false;
+			break;
+		case WORKMAN:
+			if (record->event.pressed) {
+				set_single_persistent_default_layer(_WORKMAN);
+			}
+			return false;
+			break;
+		case SBKSP:
+			if (record->event.pressed) {
+				register_code(KC_LSFT);
+				bksp = true;
+			}
+			else {
+				unregister_code(KC_LSFT);
+				if (bksp) {
+					tap_code(KC_BSPC);
+				}
+			}
+			return false;
+			break;
+        case ENDW:
+            if (record->event.pressed) {
+                register_code(KC_LCTL);
+                tap_code(KC_RGHT);
+                unregister_code(KC_LCTL);
+                tap_code(KC_LEFT);
+            }
+	}
+	bksp = false;
+	return true;
+
+}
+
+const keypos_t hand_swap_config[MATRIX_ROWS][MATRIX_COLS] = {
+	{{0, 4}, {1, 4}, {2, 4}, {3, 4}, {4, 4}, {5, 4}, {6, 4}, {7, 4}},
+	{{0, 5}, {1, 5}, {2, 5}, {3, 5}, {4, 5}, {5, 5}, {6, 5}, {7, 5}},
+	{{0, 6}, {1, 6}, {2, 6}, {3, 6}, {4, 6}, {5, 6}, {6, 6}, {7, 6}},
+	{{0, 7}, {1, 7}, {2, 7}, {3, 7}, {4, 7}, {5, 7}, {6, 7}, {7, 7}},
+	{{0, 0}, {1, 0}, {2, 0}, {3, 0}, {4, 0}, {5, 0}, {6, 0}, {7, 0}},
+	{{0, 1}, {1, 1}, {2, 1}, {3, 1}, {4, 1}, {5, 1}, {6, 1}, {7, 1}},
+	{{0, 2}, {1, 2}, {2, 2}, {3, 2}, {4, 2}, {5, 2}, {6, 2}, {7, 2}},
+	{{0, 3}, {1, 3}, {2, 3}, {3, 3}, {4, 3}, {5, 3}, {6, 3}, {7, 3}}
+};	
+
```

Comment:
Honestly, I'd rather that this be pulled out of this PR and added to the keyboard, in a separate PR.

Reply:
I'll try and do that

Reply:
Should be added in https://github.com/qmk/qmk_firmware/pull/10397 

I guess that should be merged first then I can remove the extra swap hands from my keymap?

Reply:
Should rebase this branch on `qmk:master` now.

Reply:
done!